why.js

```diff
@@ -205,6 +206,40 @@ module.exports = function whyNotEqual(value, other) {
 		if (isProto.call(other, value)) { return 'second argument is the [[Prototype]] of the first'; }
 		if (getPrototypeOf(value) !== getPrototypeOf(other)) { return 'arguments have a different [[Prototype]]'; }
 
+		var primitiveHints = [];
+		var valueIsFn = typeof value === 'function';
+		var otherIsFn = typeof other === 'function';
+		if (!valueIsFn || !otherIsFn) {
+			primitiveHints.push(Number);
+			primitiveHints.push(String);
+		}
+
+		for (var primitiveHint in primitiveHints) {
```

Comment:
for-in must never be used for iterating an array; it’s for objects. use a normal for loop.

better, would be to go back to the loopless version.